xload/providers/cached/options.go

```diff
@@ -0,0 +1,53 @@
+package cached
+
+import "time"
+
+// Option configures a cached loader.
+type Option interface {
+	apply(*options)
+}
+
+type optionFunc func(*options)
+
+func (f optionFunc) apply(opts *options) { f(opts) }
+
+// TTL sets the TTL for the cached keys.
+type TTL time.Duration
+
+func (t TTL) apply(o *options) { o.ttl = time.Duration(t) }
+
+// Cache sets the cache implementation for the loader.
+func Cache(c Cacher) Option {
+	return optionFunc(func(o *options) { o.cache = c })
+}
+
+// DisableEmptyValueHit disables caching of empty values.
+func DisableEmptyValueHit() Option {
```

Comment:
nit: Do we need it to be a function call? Can change it to a global variable also, that also adds type info. As I suggested in previous review.

Reply:
```go
// DisableEmptyValueHit disables caching of empty values.
var DisableEmptyValueHit Option = emptyValueHit(false)

type emptyValueHit bool

func (e emptyValueHit) apply(o *options) { o.emptyValueHit = bool(e) }
```

```diff
--- cachedLoader := NewLoader(loader, TTL(ttl), Cache(mc), DisableEmptyValueHit())
+++ cachedLoader := NewLoader(loader, TTL(ttl), Cache(mc), DisableEmptyValueHit)
```